yjit/src/core.rs

```diff
@@ -2298,19 +2298,35 @@ c_callable! {
     /// Generated code calls this function with the SysV calling convention.
     /// See [gen_call_entry_stub_hit].
     fn entry_stub_hit(entry_ptr: *const c_void, ec: EcPtr) -> *const u8 {
-        with_vm_lock(src_loc!(), || {
-            match with_compile_time(|| { entry_stub_hit_body(entry_ptr, ec) }) {
-                Some(addr) => addr,
-                // Failed to service the stub by generating a new block so now we
-                // need to exit to the interpreter at the stubbed location.
-                None => return CodegenGlobals::get_stub_exit_code().raw_ptr(),
-            }
+        with_compile_time(|| {
+            with_vm_lock(src_loc!(), || {
+                let cb = CodegenGlobals::get_inline_cb();
+                let ocb = CodegenGlobals::get_outlined_cb();
+
+                let addr = entry_stub_hit_body(entry_ptr, ec, cb, ocb)
+                    .unwrap_or_else(|| {
+                        // Trigger code GC (e.g. no space).
+                        // This entry point will be recompiled later.
+                        cb.code_gc(ocb);
```

Comment:
We could hit this in undesirable conditions, e.g. `stack_size` does not fit in `u8`. Can we at least check `has_dropped_bytes()` here?

Reply:
We probably should, and for all the other code gc trigger points too. But I think it should go with the `--yjit-disable-code-gc` PR.